modules/markdown/assets/js/jetpack-markdown-block-editor.jsx

```diff
@@ -0,0 +1,76 @@
+/* global React:true */
+
+/**
+ * Internal dependencies
+ */
+import MarkdownPreview from './components/markdown-preview';
+import MarkdownLivePreview from './components/markdown-live-preview';
+
+const { __ } = window.wp.i18n;
+
+const {
+	Component,
+} = window.wp.element;
+
+class JetpackMarkdownBlockEditor extends Component {
+
+	constructor() {
+		super( ...arguments );
+
+		this.updateSource = this.updateSource.bind( this );
+		this.showEditor = this.showEditor.bind( this );
+		this.showPreview = this.showPreview.bind( this );
+		this.isEmpty = this.isEmpty.bind( this );
+	}
+
+	isEmpty() {
+		const source = this.props.attributes.source;
+		return ! source || source.trim() === '';
+	}
+
+	updateSource( evt ) {
+		this.props.setAttributes( { source: evt.target.value } );
+	}
+
+	showEditor() {
+		this.setState( { activePanel: 'editor' } );
+	}
+
+	showPreview() {
+		this.setState( { activePanel: 'preview' } );
+	}
+
+	render() {
+		const { attributes, className, isSelected } = this.props;
+
+		// Renders the editor panel or the preview panel based on component's state
+		const editorOrPreviewPanel = function() {
+			const source = attributes.source;
+
+			if ( ! isSelected && ! this.isEmpty() ) {
+				return <MarkdownPreview source={ source } />;
+			}
+
+			const placeholderSource = __( 'Write your _Markdown_ **here**...' );
+
+			if ( ! isSelected && this.isEmpty() ) {
+				return <p className={ `${ className }-placeholder` }>
+					{ placeholderSource }
+				</p>;
+			}
+			return <MarkdownLivePreview
```

Comment:
I suggest tweaking the terminology somewhere. The distinction between "Preview" and "Live Preview" can be confusing.

Reply:
Totally agree. Actually, this was something I was planning to do because indeed the names are very similar and can be confusing.